master/internal/user/authz_basic_impl.go

```diff
@@ -0,0 +1,22 @@
+package user
+
+import (
+	"fmt"
+
+	"github.com/determined-ai/determined/master/pkg/model"
+)
+
+// UserAuthZBasic is basic.
+type UserAuthZBasic struct{}
+
+// CanSetUserPassword for basic authz.
+func (a *UserAuthZBasic) CanSetUserPassword(currentUser model.User, targetUser model.User) error {
+	if !currentUser.Admin && currentUser.ID != targetUser.ID {
+		return fmt.Errorf("non-admin users can only change their own password")
+	}
+	return nil
+}
+
+func init() {
+	AuthZProvider.Register("basic", &UserAuthZBasic{})
```

Comment:
how does this init pattern work if you need the config in initialization?

Reply:
and that we end up with a registry when there should only be one valid `UserAuthZ` is interesting.

Reply:
do you mean you want to initialize something inside `UserAuthZBasic`?
I think you could add `sync.Once` to `UserAuthZBasic` and then call a helper that uses it to initialize once lazily on first use (in all the implemented methods).

Reply:
I'm just reconciling this with my RM implementation, where there is a `Setup` method that must be called and it initializes a `type ResourceManager interface` singleton in the package. These things are different, but it'd be nice to me if they all followed a similar-ish pattern.

Reply:
>and that we end up with a registry when there should only be one valid UserAuthZ is interesting.

one could implement UserAuthZ which instantiates and calls two other UserAuthZ implementations, e.g. when we want "permissive rbac with audit, enforce basic" strategy.

Reply:
> I think you could add sync.Once to UserAuthZBasic and then call a helper that uses it to initialize once lazily on first use (in all the implemented methods).

yeah, that just seems less than ideal, but I suppose it would work.

Reply:
> one could implement UserAuthZ which instantiates and calls two other UserAuthZ implementations, e.g. when we want "permissive rbac with audit, enforce basic" strategy.

ah yeah, I would['ve just thought to register a new type in this case but I see.

Reply:
once initialization can fail, failing in initialization on the first call kinda sucks, though